tests/hla/test_run_svmpoller.py

```diff
@@ -0,0 +1,141 @@
+        install_dir = os.path.dirname(__file__)
+        default_file = os.path.join(install_dir, master_list)
+        if os.path.exists(default_file):
+            # Copy the file
+            shutil.copy2(default_file, os.getcwd())
+
+    # Read the master_list to get the poller fully qualified filenames
+    table = Table.read(master_list, format="ascii.fast_no_header")
+    poller_fqfile_list = table["col1"].tolist()
+
+    print("Input file: {}".format(master_list))
+    print("List of poller files: {}".format(poller_fqfile_list))
+    metafunc.parametrize('dataset', poller_fqfile_list)
+
+
+@pytest.mark.bigdata
+@pytest.mark.slow
+@pytest.mark.unit
+def test_run_svmpoller(tmpdir, dataset):
+    """ Tests to read a series of poller files and process the contents of each as Single Visit Mosaic
+
+        Characteristics of these tests:
+
+        Success Criteria:
+            The SVM processing returns a value of 0: Success or 1: Failure
+
+        The input master_list file is a list of poller filenames, one filename per line.
+        Each poller file must be obtained from a specified directory and read to obtain the
+        names of the data files which need to be processed.
+
+        This test file can be executed in the following manner:
+            $ pytest -n # -s --basetemp=/internal/hladata/yourUniqueDirectoryHere --bigdata --slow
+              --master_list /internal/hladata/input/master_poller_list.txt test_run_svmpoller.py >&
+              test_svmpoller_output.txt &
+            $ tail -f test_svmpoller_output.txt
+          * The `-n #` option can be used to run tests in parallel if `pytest-xdist` has
+            been installed where `#` is the number of cpus to use. THIS IS NOT ADVISED FOR USE.
+          * Note: When running this test, the `--basetemp` directory should be set to a unique
+            existing directory to avoid deleting previous test output.
+          * A default master list exists in the tests/hla directory and contains 121 datasets.  This
+            is probably NOT the list you want to use, but it allows you to see what this file should
+            contain.
+
+    """
+    print("TEST_RUN_SVMPOLLER. Dataset: ", dataset)
+    output_name = dataset + '.ecsv'
+
+    current_dt = datetime.datetime.now()
+    print(str(current_dt))
+
+    subdir = ""
+    prevdir = os.getcwd()
+
+    # create working directory specified for the test
+    if not tmpdir.ensure(subdir, dir=True):
+        cpyurdir = tmpdir.mkdir(subdir).strpath
+    else:
+        curdir = tmpdir.join(subdir).strpath
+    os.chdir(curdir)
+
+    return_value = 1
+
+    try:
+
+        # The dataset variable is a fully qualified poller file (CSV) with no header
+        # Copy the poller file to the current working directory
+        shutil.copy2(dataset, ".")
+
+        # Get the poller file path, as well as its simple name
+        poller_path = "/".join(dataset.split("/")[:-1])
+        poller_file = dataset.split("/")[-1]
```

Comment:
It is usually better to be a little more platform agnostic when parsing filenames and split on 'os.sep' instead of the magic literal '/'.  In this case, the tests all run in the same type of Linux environment, but that may not always be the case.

Reply:
Using os methods to be OS independent.